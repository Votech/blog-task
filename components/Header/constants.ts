export const TEXT = {
  BLOG: 'blog',
};
