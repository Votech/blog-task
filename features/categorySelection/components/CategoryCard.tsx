import { PostCategory } from '@/types/blog';
import clsx from 'clsx';
import Image from 'next/image';

interface Props {
  category: PostCategory;
  onSelectCategory: (category: PostCategory) => void;
  imgSrc: string;
  Icon: React.ComponentType | null;
  textColor: string;
  bgColor: string;
  borderColor: string;
  selected: boolean;
}

export default function CategoryCard({
  category,
  onSelectCategory,
  imgSrc,
  Icon,
  bgColor = 'bg-category-green',
  borderColor = 'bg-category-green-dark',
  textColor = 'text-category-green',
  selected = true,
}: Props) {
  function handleClick() {
    onSelectCategory(category);
  }

  return (
    <div
      onClick={handleClick}
      className={clsx(
        'rounded-diagonal-lg relative flex h-full min-h-72 w-60 w-full cursor-pointer flex-col items-center overflow-hidden xs:w-auto sm:min-h-0',
        bgColor,
      )}
    >
      {/* Overlay border */}
      {selected && (
        <div
          className={clsx(
            'rounded-diagonal-lg absolute inset-0 border-[6px]',
            borderColor,
          )}
        />
      )}
      <Image src={imgSrc} alt={category} width={366} height={230}></Image>
      <div className="flex flex-col items-center gap-4 p-6">
        <p className={clsx('font-bold', textColor)}>{category.toUpperCase()}</p>
        {Icon && <Icon />}
      </div>
    </div>
  );
}
