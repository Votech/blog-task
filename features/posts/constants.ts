export const POSTS_PER_PAGE = 50;
export const POSTS_LIMIT = 100;
