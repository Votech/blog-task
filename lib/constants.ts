import { PostCategory } from '@/types/blog';

export const CATEGORY = {
  knowledge: 'wiedza',
  inspirations: 'inspiracje',
  interpretations: 'interpretacje',
  available: 'dostÄ™pne',
} as const;

export const CATEGORY_LIST: PostCategory[] = Object.values(CATEGORY);

export const LIKED_POSTS_KEY = 'likedPosts';

export const CATEGORY_STYLES: Record<
  PostCategory,
  {
    background: string;
    borderColor: string;
    textColor: string;
    highlightColor: string;
  }
> = {
  [CATEGORY.knowledge]: {
    background: 'bg-category-blue',
    borderColor: 'border-category-blue-dark',
    textColor: 'text-white',
    highlightColor: 'text-category-blue',
  },
  [CATEGORY.inspirations]: {
    background: 'bg-category-orange',
    borderColor: 'border-category-orange-dark',
    textColor: 'text-silver-grey',
    highlightColor: 'text-category-orange',
  },
  [CATEGORY.interpretations]: {
    background: 'bg-category-red',
    borderColor: 'border-category-red-dark',
    textColor: 'text-white',
    highlightColor: 'text-category-red',
  },
  [CATEGORY.available]: {
    background: 'bg-category-green',
    borderColor: 'border-category-green-dark',
    textColor: 'text-cold-blue',
    highlightColor: 'text-category-green',
  },
};
