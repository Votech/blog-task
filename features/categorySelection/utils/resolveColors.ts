import { PostCategory } from '@/types/blog';
import { CATEGORY } from '../constants';

export default function resolveColors(category: PostCategory) {
  const colors = {
    [CATEGORY.knowledge]: {
      bg: 'bg-category-blue',
      border: 'border-category-blue-dark',
      text: 'text-white',
    },
    [CATEGORY.inspirations]: {
      bg: 'bg-category-orange',
      border: 'border-category-orange-dark',
      text: 'text-silver-grey',
    },
    [CATEGORY.interpretations]: {
      bg: 'bg-category-red',
      border: 'border-category-red-dark',
      text: 'text-white',
    },
    [CATEGORY.available]: {
      bg: 'bg-category-green',
      border: 'border-category-green-dark',
      text: 'text-cold-blue',
    },
    default: {
      bg: 'bg-category-green',
      border: 'border-category-green-dark',
      text: 'text-category-green',
    },
  };

  return colors[category] || colors.default;
}
