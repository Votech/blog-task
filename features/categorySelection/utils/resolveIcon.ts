import { PostCategory } from '@/types/blog';
import dynamic from 'next/dynamic';
import { CATEGORY } from '../../../lib/constants';

const BookSvg = dynamic(() => import('../icons/Book'));
const CloudToolsSvg = dynamic(() => import('../icons/CloudTools'));
const GlassesSvg = dynamic(() => import('../icons/Glasses'));
const IdeaSvg = dynamic(() => import('../icons/Idea'));

export default function resolveIcon(category: PostCategory) {
  switch (category) {
    case CATEGORY.knowledge:
      return BookSvg;
    case CATEGORY.inspirations:
      return IdeaSvg;
    case CATEGORY.interpretations:
      return CloudToolsSvg;
    case CATEGORY.available:
      return GlassesSvg;
    default:
      return null;
  }
}
