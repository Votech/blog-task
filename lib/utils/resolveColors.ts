import { CATEGORY } from '@/lib/constants';
import { PostCategory } from '@/types/blog';

export default function resolveColors(category: PostCategory) {
  const colors = {
    [CATEGORY.knowledge]: {
      bg: 'bg-category-blue',
      border: 'border-category-blue-dark',
      text: 'text-white',
      textColor: 'text-category-blue',
    },
    [CATEGORY.inspirations]: {
      bg: 'bg-category-orange',
      border: 'border-category-orange-dark',
      text: 'text-silver-grey',
      textColor: 'text-category-orange',
    },
    [CATEGORY.interpretations]: {
      bg: 'bg-category-red',
      border: 'border-category-red-dark',
      text: 'text-white',
      textColor: 'text-category-red',
    },
    [CATEGORY.available]: {
      bg: 'bg-category-green',
      border: 'border-category-green-dark',
      text: 'text-cold-blue',
      textColor: 'text-category-green',
    },
    default: {
      bg: 'bg-category-green',
      border: 'border-category-green-dark',
      text: 'text-category-green',
      textColor: 'text-category-green',
    },
  };

  return colors[category] || colors.default;
}
