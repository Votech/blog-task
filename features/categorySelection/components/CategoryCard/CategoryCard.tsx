import resolveColors from '@/lib/utils/resolveColors';
import { PostCategory } from '@/types/blog';
import clsx from 'clsx';
import Image from 'next/image';
import resolveIcon from '../../utils/resolveIcon';

interface Props {
  cardCategory: PostCategory;
  onSelectCategory: (category: PostCategory) => void;
  selected: boolean;
  imagePriority?: boolean;
}

export default function CategoryCard({
  cardCategory,
  onSelectCategory,
  selected,
  imagePriority = true,
}: Props) {
  const { background, borderColor, textColor } = resolveColors(cardCategory);
  const Icon = resolveIcon(cardCategory);
  const imgSrc = `/images/${cardCategory}.webp`;

  return (
    <div
      onClick={() => onSelectCategory(cardCategory)}
      className={clsx(
        'rounded-diagonal-lg relative flex h-full min-h-72 w-60 w-full cursor-pointer flex-col items-center overflow-hidden xs:w-auto sm:min-h-0',
        background,
      )}
    >
      {/* Overlay border */}
      {selected && (
        <div
          className={clsx(
            'rounded-diagonal-lg absolute inset-0 border-[6px]',
            borderColor,
          )}
        />
      )}
      <Image
        priority={imagePriority}
        loading={imagePriority ? 'eager' : 'lazy'}
        src={imgSrc}
        alt={cardCategory}
        width={366}
        height={230}
      />
      <div className="flex flex-col items-center gap-4 p-6">
        <p className={clsx('font-bold', textColor)}>
          {cardCategory.toUpperCase()}
        </p>
        {Icon && <Icon />}
      </div>
    </div>
  );
}
