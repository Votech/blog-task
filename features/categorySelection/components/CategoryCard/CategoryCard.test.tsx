import { CATEGORY } from '@/lib/constants';
import resolveColors from '@/lib/utils/resolveColors';
import { render } from '@testing-library/react';
import resolveIcon from '../../utils/resolveIcon';
import CategoryCard from './CategoryCard';

jest.mock('@/lib/utils/resolveColors', () => jest.fn());
jest.mock('../../utils/resolveIcon', () => jest.fn());

const mockOnSelectCategory = jest.fn();
const mockCategory = CATEGORY.knowledge;
const mockColors = {
  background: 'bg-category-blue',
  borderColor: 'border-category-blue-dark',
  textColor: 'text-white',
  highlightColor: 'text-category-blue',
};
const MockIcon = () => <svg data-testid="mock-icon" />;

describe('CategoryCard Snapshot', () => {
  beforeEach(() => {
    (resolveColors as jest.Mock).mockReturnValue(mockColors);
    (resolveIcon as jest.Mock).mockReturnValue(MockIcon);
  });

  it('matches snapshot when selected is false', () => {
    const { asFragment } = render(
      <CategoryCard
        cardCategory={mockCategory}
        onSelectCategory={mockOnSelectCategory}
        selected={false}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when selected is true', () => {
    const { asFragment } = render(
      <CategoryCard
        cardCategory={mockCategory}
        onSelectCategory={mockOnSelectCategory}
        selected={true}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
