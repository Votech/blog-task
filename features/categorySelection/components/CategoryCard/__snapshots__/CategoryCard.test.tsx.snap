// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Snapshot matches snapshot when selected is false 1`] = `
<DocumentFragment>
  <div
    class="rounded-diagonal-lg relative flex h-full min-h-72 w-60 w-full cursor-pointer flex-col items-center overflow-hidden xs:w-auto sm:min-h-0 bg-category-blue"
  >
    <img
      alt="wiedza"
      data-nimg="1"
      decoding="async"
      height="230"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fwiedza.webp&w=750&q=75"
      srcset="/_next/image?url=%2Fimages%2Fwiedza.webp&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fwiedza.webp&w=750&q=75 2x"
      style="color: transparent;"
      width="366"
    />
    <div
      class="flex flex-col items-center gap-4 p-6"
    >
      <p
        class="font-bold text-white"
      >
        WIEDZA
      </p>
      <svg
        data-testid="mock-icon"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`CategoryCard Snapshot matches snapshot when selected is true 1`] = `
<DocumentFragment>
  <div
    class="rounded-diagonal-lg relative flex h-full min-h-72 w-60 w-full cursor-pointer flex-col items-center overflow-hidden xs:w-auto sm:min-h-0 bg-category-blue"
  >
    <div
      class="rounded-diagonal-lg absolute inset-0 border-[6px] border-category-blue-dark"
    />
    <img
      alt="wiedza"
      data-nimg="1"
      decoding="async"
      height="230"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fwiedza.webp&w=750&q=75"
      srcset="/_next/image?url=%2Fimages%2Fwiedza.webp&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fwiedza.webp&w=750&q=75 2x"
      style="color: transparent;"
      width="366"
    />
    <div
      class="flex flex-col items-center gap-4 p-6"
    >
      <p
        class="font-bold text-white"
      >
        WIEDZA
      </p>
      <svg
        data-testid="mock-icon"
      />
    </div>
  </div>
</DocumentFragment>
`;
